### PR TITLE
Pass ``doc_node`` to postinit of all ``PartialFunction`` constructors

### PR DESCRIPTION
**Pull request by DanielNoord**
## Steps

- [x] Write a good description on what the PR does.

## Description

Similar to #1436.

Partial step 2 as discussed in https://github.com/PyCQA/astroid/pull/1432#pullrequestreview-898319366.

Blocked by https://github.com/PyCQA/astroid/pull/1434 as there is a test that depends on the deprecation.
Note: there is currently no test of `PartialFunction` itself, which also showed that we don't pass `end_lineno` etc. to the constructor. That is out of scope for this PR though.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
| ✓   | :hammer: Refactoring   |

### CODE REVIEW

File: astroid/brain/brain_functools.py [hunk 1]
```diff
@@ -101,7 +105,7 @@ def _functools_partial_inference(node, context=None):
     partial_function = objects.PartialFunction(
         call,
         name=inferred_wrapped_function.name,
-        doc=inferred_wrapped_function.doc,
+        doc=getattr(inferred_wrapped_function.doc_node, "value", None),
```
Comment (cdce8p) [inline astroid/brain/brain_functools.py:108]:
```suggestion
```

File: tests/unittest_brain.py [hunk 2]
```diff
@@ -2994,6 +2994,29 @@ def test_infer_dict_from_keys() -> None:
...
+            return a + b
+        partial(test, 1)(3) #@
+        """
+        )
+        assert isinstance(ast_node.func, nodes.Call)
+        inferred = ast_node.func.inferred()
+        assert len(inferred) == 1
+        partial = inferred[0]
+        assert isinstance(partial, objects.PartialFunction)
+        assert partial.doc_node
```
Comment (cdce8p) [inline tests/unittest_brain.py:3012]:
```suggestion
        assert isinstance(partial.doc_node, nodes.Const)
```

File: tests/unittest_brain.py [hunk 3]
```diff
@@ -2994,6 +2994,29 @@ def test_infer_dict_from_keys() -> None:
 
 
 class TestFunctoolsPartial:
+    def test_infer_partial(self) -> None:
```
Comment (cdce8p) [inline tests/unittest_brain.py:2997]: This can be a `staticmethod`. `self` isn't used.

**Review by cdce8p — CHANGES_REQUESTED** [review]

File: astroid/brain/brain_functools.py (hunk 1 above)
Comment (DanielNoord) [reply astroid/brain/brain_functools.py:108]: This depends on order of deprecation. See other comments in other PRs.

File: tests/unittest_brain.py (hunk 3 above)
Comment (DanielNoord) [reply tests/unittest_brain.py]: I got used to `pylint` warning me about this. Forgot it is turned off for `astroid`.

Comment (cdce8p) [reply tests/unittest_brain.py]: Sometimes it's helpful, but not always. I opened an issue in pylint a while back: https://github.com/PyCQA/pylint/issues/5502

**Review by cdce8p — APPROVED** [review]

**Pull request merged**